tests/ui/pattern/deref-patterns/implicit-cow-deref.rs

```diff
@@ -0,0 +1,44 @@
+//@ run-pass
+//! Test that implicit deref patterns interact as expected with `Cow` constructor patterns.
+#![feature(deref_patterns)]
+#![allow(incomplete_features)]
+
+use std::borrow::Cow;
+
+fn main() {
+    let cow: Cow<'static, [u8]> = Cow::from(&[1, 2, 3]);
```

Comment:
I had to look at the docs to check which one that was x)
```suggestion
    let cow: Cow<'static, [u8]> = Cow::Borrowed(&[1, 2, 3]);
```